### PR TITLE
api/vk: use proper api, add support for more links & access keys, refactor

### LINKED ISSUE #956: VK Video support is broken
### bug description

Couple days ago VK registered their new domain "vkvideo.ru" for videos published on VK, and since now it redirects all videos to new domain, and the support of VK is broken.

Example link: 

### reproduction steps

1. paste new link, press dl
2. _your link is invalid or this service is not supported yet. have you pasted the right link?_

1. paste link with vkvideo.ru replaced to vk.com, press dl
2. _couldn't find any media that i could download for you. are you sure you pasted the right link?_

### screenshots

_No response_

### links

```shell
https://vkvideo.ru/video-4429782_76849466
https://vkvideo.ru/video-1719791_456263870
https://vkvideo.ru/video-25016515_456240870
```

### platform information

Zen Browser 1.0.1-a.19 (Based on Firefox 132.0.1) on Windows 11

### additional context

_No response_

### PR DESCRIPTION
**Pull request by wukko**
closes #956

### CODE REVIEW

File: api/src/processing/services/vk.js [hunk 1]
```diff
@@ -1,62 +1,144 @@
...
+    if (!video.files || !video.duration) {
+        return { error: "fetch.fail" };
     }
 
-    if (js.mvData.duration > env.durationLimit) {
+    if (video.duration > env.durationLimit) {
         return { error: "content.too_long" };
     }
 
+    const userQuality = quality === "max" ? 2160 : quality;
```
Comment (dumbmoron) [inline api/src/processing/services/vk.js:110]:
```suggestion
    const userQuality = quality === "max" ? resolutions[0] : quality;
```

:3c ?

File: api/src/processing/services/vk.js [hunk 2]
```diff
@@ -1,62 +1,144 @@
...
+    let pickedQuality;
+
     for (let i in resolutions) {
```
Comment (dumbmoron) [inline api/src/processing/services/vk.js:113]:
```suggestion
    for (const resolution of resolutions) {
```

File: api/src/processing/services/vk.js [hunk 3]
```diff
@@ -1,62 +1,144 @@
...
-        if (js.player.params[0][`url${resolutions[i]}`]) {
-            quality = resolutions[i];
+        if (video.files[`mp4_${resolutions[i]}`]) {
+            pickedQuality = resolutions[i];
```
Comment (dumbmoron) [inline api/src/processing/services/vk.js:115]:
```suggestion
        if (video.files[`mp4_${resolution}`] && +resolution <= +userQuality) {
            pickedQuality = resolution;
```

File: api/src/processing/services/vk.js [hunk 4]
```diff
@@ -1,62 +1,144 @@
...
             break
         }
     }
-    if (Number(quality) > Number(o.quality)) quality = o.quality;
 
-    url = js.player.params[0][`url${quality}`];
+    if (Number(pickedQuality) > Number(userQuality)) {
+        pickedQuality = userQuality;
+    }
+
```
Comment (dumbmoron) [inline api/src/processing/services/vk.js:123]:
```suggestion
```

**Review by dumbmoron — APPROVED** [review]
lgtj :3

**Comment by wukko** [comment]
we merge /fast-forward

**Pull request merged**